library/psa_crypto.c

```diff
@@ -5728,7 +5728,7 @@ static void psa_des_set_key_parity(uint8_t *data, size_t data_size)
  * Note: Function allocates memory for *data buffer, so given *data should be
  *       always NULL.
  */
-#if defined(MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY) || \
+#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE) || \
```

Comment:
Derivation involves private keys and cannot be done with only public key support. So I agree that checking `PUBLIC_KEY` here didn't make sense.

I don't understand why `|| BUILTIN` either. What could make sense is **`&& BUILTIN`** (or just checking `BUILTIN` and not `WANT`) because we haven't implemnted the driver interface for key derivation yet. I don't know if that's necessary: maybe this bit can work independently of the rest of the driver interface. But if it doesn't then checking `BUILTIN` would make sense as a temporary limitation.

Reply:
Wdyt if I set the following guards:

```
defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE) ||
defined(MBEDTLS_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_DERIVE)
```

The `PSA_WANT` part should be "temporary" until key derivation is added to driver. Does it makes sense to you?

Reply:
Makes sense to me, but I think there should a comment explaining that the `WANT` part is temporary. Or actually, perhaps just use the `WANT` part with a comment that says "this should be BUILTIN really but we don't have driver support for this yet"?